cov.el

```diff
@@ -173,32 +167,39 @@ If `cov-coverage-file' is non nil, the value of that variable is returned. Other
   (or cov-coverage-file
       (setq cov-coverage-file (cov--locate-coverage (f-this-file)))))
 
-(defun cov--keep-line? (line)
-  (s-matches? cov-line-re line))
-
 (defun cov--read (file-path)
-  "Read a gcov file, filter unused lines, and return a list of lines"
-  (cl-remove-if-not
-   'cov--keep-line?
-   (cov--read-lines file-path)))
-
-(defun cov--parse (string)
-  "Returns a list of (line-num, times-ran)"
-  `(,(string-to-number (nth 2 (s-match cov-line-re string)))
-    ,(string-to-number (nth 1 (s-match cov-line-re string)))))
-
-(defun cov-make-overlay (line fringe help)
+  "Read a gcov file as buffer and return that buffer."
+  (find-file-noselect file-path))
+
+(defun cov--parse (cov-buffer)
+  "Parse a buffer containing a gcov file, filter unused lines, and return a list of (LINE-NUM TIMES-RAN)"
+  (let ((more t)
+        matches)
+    (save-excursion
+      (with-current-buffer cov-buffer
+        (save-restriction
+          (widen)
+          (goto-char (point-min)) ;; needed in case the file is already open.
```

Comment:
``` lisp
(with-temp-buffer (insert-file-contents file-path) ...) 
```
fixes this as well. No need to worry about narrowing or cursor position in temp buffers.